### PR TITLE
docs(contribute): rewrite flyte contribute docs based on 4960

### PR DESCRIPTION
**Pull request by bearomorphism**
## Tracking issue

NA

## Why are the changes needed?

The contribution guide of `flyte` can be improved.

## What changes were proposed in this pull request?

Rewrote the contribution guide of `flyte`.

## Check all the applicable boxes 

- [x] I updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Related PRs

https://github.com/flyteorg/flyte/pull/4960

### CODE REVIEW

**Review by MortalHappiness — DISMISSED** [review]

**Review by jasonlai1218 — DISMISSED** [review]
LGTM

File: docs/community/contribute.rst [hunk 1]
```diff
@@ -153,12 +153,28 @@ To understand how the below components interact with each other, refer to :ref:`
     * - **Purpose**: Deployment, Documentation, and Issues
     * - **Languages**: RST
 
-To build the Flyte docs locally you will need the following prerequisites:
+In the ``flyteorg/flyte`` root directory you can run ``make dev-docs`` to build the documentation locally. The generated documentation will be in the ``docs/_build/html`` directory.
 
-* Install `conda-lock <https://github.com/conda/conda-lock>`__.
```
Comment (neverett) [inline docs/community/contribute.rst:158]: Can we leave these steps in as an alternative to the steps below? You can move them down and put them in a "Alternative conda-lock setup steps" section.

Comment (bearomorphism) [reply docs/community/contribute.rst:158]: I can add the following steps back as an alternative.

https://github.com/flyteorg/flyte/blob/84c0fefef78ad603f26f40ea0ae2cac4d2c32711/docs/community/contribute.rst?plain=1#L156-L164

Comment (bearomorphism) [reply docs/community/contribute.rst:158]: Done.

Comment (neverett) [reply docs/community/contribute.rst:158]: Thanks!

**Review by neverett — APPROVED** [review]

**Comment by Mecoli1219** [comment]
@bearomorphism  This PR is really helpful! I followed the old document build guides on my M2 Macos and got really frustrated. Could someone help push this PR to improve the current contribution guide?

**Comment by Future-Outlier** [comment]
Hi, @bearomorphism 
We can get this merged, can you solve the conflict problem?

File: docs/community/contribute/contribute_code.rst [hunk 2]
```diff
@@ -25,12 +25,38 @@ To understand how the below components interact with each other, refer to :ref:`
     * - **Purpose**: Deployment, Documentation, and Issues
     * - **Languages**: RST
 
-To build the Flyte docs locally you will need the following prerequisites:
+In the ``flyteorg/flyte`` root directory you can run ``make dev-docs`` to build the documentation locally. The generated documentation will be in the ``docs/_build/html`` directory.
 
+**Setup process**
+
+1. First you need to make sure you can run linux/amd64 container
```
Comment (wild-endeavor) [inline docs/community/contribute/contribute_code.rst:32]: Can arm macs by default run amd64 images?  Not well right?  I feel like I get errors all the time.  Is the image multi-arch?

**Review by wild-endeavor — APPROVED** [review]
We can merge this, but can we add at some point instructions that don't involve conda?  I like using uv, and i think it's plenty fast.  If conda is not required, can we reword "We recommend Miniconda..."

**Pull request merged**